compiler/src/dotty/tools/dotc/transform/patmat/Space.scala

```diff
@@ -84,6 +84,9 @@ trait SpaceLogic {
   /** Is `tp1` the same type as `tp2`? */
   def isEqualType(tp1: Type, tp2: Type): Boolean
 
+  /** Return a space containing the values of both atom (non-decomposable) types. */
+  def intersectAtomTypes(tp1: Type, tp2: Type): Space
```

Comment:
Better to add one additional note saying that `tp1` and `tp2` are not subtype of each other.